### PR TITLE
Default upgrade

### LINKED ISSUE #6683: Fix default version settings for operator
**User Story**

The default version settings used by the operator in master branch and `release/v2.16` have the following issues:
* Automatic update from 1.15 should not have a `*` in the `To` field as it is not supported but refer to a fixed version (e.g. 1.6.13). See https://github.com/kubermatic/kubermatic/blob/b115d108db0c3bd93f7511ee38ee0c05c53ff2d2/pkg/version/manager.go#L190-L194
* We shave automatic updates from 1.16 to 1.17 as it is deprecated.

https://github.com/kubermatic/kubermatic/blob/6f6157c6304e0ed37205472e0e57e15279fe1283/pkg/controller/operator/common/defaults.go#L214-L237

**Acceptance criteria**

The default version settings of the operator trigger automatic migration to 1.15 an 1.16 to 1.17.

### PR DESCRIPTION
**Pull request by emmanuel-kubermatic**
**What this PR does / why we need it**:
The PR fixes #6683 and allows default upgrades on startup

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6683

**Special notes for your reviewer**:

**Documentation**:

https://github.com/kubermatic/kubermatic/blob/master/docs/dev-getting-started.md

**Does this PR introduce a user-facing change?**:
```release-note
Fix default version configuration to have automatic upgrade from Kubernetes 1.16 to 1.17.
```

### CODE REVIEW

File: pkg/controller/operator/common/defaults.go [hunk 1]
```diff
@@ -231,7 +228,8 @@ var (
 			{
 				// Allow to next minor release
```
Comment (irozzo-1A) [inline pkg/controller/operator/common/defaults.go]: Nit: I would put:
```
// Auto-upgrade unsupported clusters
```

**Comment by irozzo-1A** [comment]
@emmanuel-kubermatic Overall LGTM. In order to fix the failing tests you should run:

```
hack/update-docs.sh
hack/update-kubermatic-chart.sh
```

and:

```
gofmt -s -w pkg/controller/operator/common/defaults.go
```

**Comment by emmanuel-kubermatic** [comment]
Thanks a ton !

File: pkg/controller/operator/common/defaults.go [hunk 2]
```diff
@@ -190,6 +190,8 @@ var (
 	DefaultKubernetesVersioning = operatorv1alpha1.KubermaticVersioningConfiguration{
 		Default: semver.MustParse("v1.19.3"),
 		Versions: []*semver.Version{
+			// Kubernetes 1.16
+			semver.MustParse("v1.16.13"),
```
Comment (irozzo-1A) [inline pkg/controller/operator/common/defaults.go]: Mmm, I was thinking that putting back a `v1.16` version just to put the auto-upgrade from `1.15` to `1.16` does not seem to be a good option. This version will be visible to users. I think that for the long term we should rework the auto-upgrade mechanism to cover the case where the destination minor version is not supported, but for the moment I would just remove this and the auto-upgrade `1.15` -> `1.16`, customers that still have 1.15 clusters will need to override the default settings to be able to upgrade. WDYT?

Comment (emmanuel-kubermatic) [reply pkg/controller/operator/common/defaults.go]: That's fine

If that cluster version is Out Of Life, no point for the customer using it

File: charts/kubermatic/static/master/versions.yaml [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # This file has been generated using hack/update-kubermatic-chart.sh, do not edit.
 
 versions:
+- type: kubernetes
+  version: 1.16.13
```
Comment (irozzo-1A) [inline charts/kubermatic/static/master/versions.yaml]: Remember to remove this too, and to regenerate the charts ;-)

File: pkg/controller/operator/common/defaults.go (hunk 2 above)
Comment (irozzo-1A) [reply pkg/controller/operator/common/defaults.go]: Ok, you can remove this version then

Comment (emmanuel-kubermatic) [reply pkg/controller/operator/common/defaults.go]: I've removed those and updating the docs for the next push

**Comment by emmanuel-kubermatic** [comment]
Done

**Pull request merged**